src/components/victory-area/victory-area.js

```diff
@@ -179,9 +179,19 @@ export default class VictoryArea extends React.Component {
     ]),
     /**
      * The samples prop specifies how many individual points to plot when plotting
+     * @param {Object} props The component of which the samples prop belongs to.
+     * @param {String} propName The name of the prop of which the function belongs to.
+     * @param {String} componentName The name of the component.
+     * @returns {String} Returns a warning to the user in the form of a String via the console.
      * y as a function of x. Samples is ignored if x props are provided instead.
      */
-    samples: CustomPropTypes.nonNegative,
+    samples: (props, propName, componentName) => {
+      if (props.x !== undefined) {
```

Comment:
This looks good but I would prefer it to be extracted into a custom prop type in victory-core.